tests/nnstreamer_filter_single/unittest_filter_single.cc

```diff
@@ -0,0 +1,285 @@
+/**
+ * @file        unittest_filter_single.cc
+ * @date        30 Aug 2021
+ * @brief       Unit test for tensor_filter_single
+ * @see         https://github.com/nnstreamer/nnstreamer
+ * @author      Jaeyun Jung <jy1210.jung@samsung.com>
+ * @bug         No known bugs
+ */
+
+#include <gtest/gtest.h>
+#include <glib.h>
+#include <nnstreamer_plugin_api.h>
+
+#include "../gst/nnstreamer/tensor_filter/tensor_filter_single.h"
+
+/**
+ * @brief Internal data structure to run tensor-filter single.
+ */
+typedef struct
+{
+  GTensorFilterSingle *single;
+  GTensorFilterSingleClass *klass;
+  GstTensorMemory input;
+  GstTensorMemory output;
+} single_test_data_s;
+
+#ifdef ENABLE_TENSORFLOW_LITE
+/**
+ * @brief Get max score from output.
+ */
+static gsize
+get_max_score (GstTensorMemory * output)
+{
+  guint8 *array;
+  gsize idx, max_idx = 0;
+  guint8 max_value = 0;
+
+  array = (guint8 *) output->data;
+
+  for (idx = 0; idx < output->size; idx++) {
+    if (max_value < array[idx]) {
+      max_idx = idx;
+      max_value = array[idx];
+    }
+  }
+
+  return max_idx;
+}
+
+/**
+ * @brief Initialize test data, load tf-lite model and raw data.
+ */
+static gboolean
+init_test_data (single_test_data_s * test_data)
+{
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  gchar *model_file, *data_file;
+  gsize length;
+  gboolean loaded = FALSE;
+
+  model_file = data_file = NULL;
+  test_data->input.size = 3U * 224 * 224;
+  test_data->output.size = 1001U;
+
+  test_data->single = (GTensorFilterSingle *) g_object_new (G_TYPE_TENSOR_FILTER_SINGLE, NULL);
+  test_data->klass = (GTensorFilterSingleClass *) g_type_class_ref (G_TYPE_TENSOR_FILTER_SINGLE);
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  model_file = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v1_1.0_224_quant.tflite", NULL);
+  if (!g_file_test (model_file, G_FILE_TEST_EXISTS)) {
+    goto error;
+  }
+
+  data_file = g_build_filename (
+      root_path, "tests", "test_models", "data", "orange.raw", NULL);
+  if (!g_file_get_contents (data_file, (gchar **) &test_data->input.data, &length, NULL) ||
+      length != test_data->input.size) {
+    goto error;
+  }
+
+  g_object_set (G_OBJECT (test_data->single), "framework", "tensorflow-lite", "model", model_file, NULL);
+  loaded = TRUE;
+
+error:
+  g_free (model_file);
+  g_free (data_file);
+
+  return loaded;
+}
+
+/**
+ * @brief Free test data.
+ */
+static void
+free_test_data (single_test_data_s * test_data)
+{
+  g_type_class_unref (test_data->klass);
+  g_object_unref (test_data->single);
+  g_free (test_data->input.data);
+  g_free (test_data->output.data);
+}
+
+/**
+ * @brief Test to invoke tf-lite model.
+ */
+TEST (testTensorFilterSingle, invoke)
+{
+  single_test_data_s tdata = { 0, };
+
+  EXPECT_TRUE (init_test_data (&tdata));
+
+  /* invoke the model and check label 'orange' (index 951) */
```

Comment:
Question. 
Should we call `tdata.klass->start (tdata.single)` before invocation?

Reply:
Internally filter-single calls start() in invoke() function. Also, when finalize() called, it calls stop().
To check this condition, I removed start() and stop() in this testcase.